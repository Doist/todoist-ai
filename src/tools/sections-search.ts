import type { Section } from '@doist/todoist-api-typescript'
import { z } from 'zod'
import { getToolOutput } from '../mcp-helpers.js'
import type { TodoistTool } from '../todoist-tool.js'
import { summarizeList } from '../utils/response-builders.js'
import { ToolNames } from '../utils/tool-names.js'

const { SECTIONS_MANAGE, TASKS_LIST_FOR_CONTAINER, TASKS_UPDATE_MULTIPLE } = ToolNames

const ArgsSchema = {
    projectId: z.string().min(1).describe('The ID of the project to search sections in.'),
    search: z
        .string()
        .optional()
        .describe(
            'Search for a section by name (partial and case insensitive match). If omitted, all sections in the project are returned.',
        ),
}

type SectionSummary = {
    id: string
    name: string
}

const sectionsSearch = {
    name: ToolNames.SECTIONS_SEARCH,
    description: 'Search for sections by name or other criteria in a project.',
    parameters: ArgsSchema,
    async execute(args, client) {
        const { results } = await client.getSections({
            projectId: args.projectId,
        })
        const searchLower = args.search ? args.search.toLowerCase() : undefined
        const filtered = searchLower
            ? results.filter((section: Section) => section.name.toLowerCase().includes(searchLower))
            : results

        const sections = filtered.map((section) => ({
            id: section.id,
            name: section.name,
        }))

        const textContent = generateTextContent({
            sections,
            projectId: args.projectId,
            search: args.search,
        })

        return getToolOutput({
            textContent,
            structuredContent: {
                sections,
                totalCount: sections.length,
                appliedFilters: args,
            },
        })
    },
} satisfies TodoistTool<typeof ArgsSchema>

function generateTextContent({
    sections,
    projectId,
    search,
}: {
    sections: SectionSummary[]
    projectId: string
    search?: string
}): string {
    const zeroReasonHints: string[] = []

    if (search) {
        zeroReasonHints.push('Try broader search terms')
        zeroReasonHints.push('Check spelling')
        zeroReasonHints.push('Remove search to see all sections')
    } else {
        zeroReasonHints.push('Project has no sections yet')
        zeroReasonHints.push(`Use ${SECTIONS_MANAGE} to create sections`)
    }

    // Data-driven next steps based on results
    const nextSteps: string[] = []

    if (sections.length > 0) {
        // Suggestions based on number of sections found
        if (sections.length === 1) {
            const sectionId = sections[0]?.id
            nextSteps.push(
                `Use ${TASKS_LIST_FOR_CONTAINER} with type=section and id=${sectionId} to see tasks`,
            )
            nextSteps.push(`Use ${SECTIONS_MANAGE} to create additional sections for organization`)
        } else if (sections.length > 8) {
            nextSteps.push(
                'Consider consolidating sections - many small sections can reduce productivity',
            )
            nextSteps.push(`Use ${TASKS_UPDATE_MULTIPLE} to move tasks between sections`)
            nextSteps.push('Use delete-one with type=section to delete empty sections')
        } else {
            nextSteps.push(
                `Use ${TASKS_LIST_FOR_CONTAINER} with type=section to see tasks in any section`,
            )
            nextSteps.push(`Use ${SECTIONS_MANAGE} to modify section names or order`)
        }

        // Search-specific suggestions
        if (search) {
            nextSteps.push('Remove search parameter to see all sections in this project')
        }
    } else {
        // Empty result suggestions are already handled in zeroReasonHints
        // No additional nextSteps needed for empty results
    }

    const subject = search
        ? `Sections in project ${projectId} matching "${search}"`
        : `Sections in project ${projectId}`

    const previewLines =
        sections.length > 0
            ? sections.map((section) => `    ${section.name} â€¢ id=${section.id}`).join('\n')
            : undefined

    return summarizeList({
        subject,
        count: sections.length,
        previewLines,
        zeroReasonHints,
        nextSteps,
    })
}

export { sectionsSearch }
