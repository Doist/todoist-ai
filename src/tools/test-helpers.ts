import type { PersonalProject, Section, Task } from '@doist/todoist-api-typescript'

/**
 * Mapped task type matching the output of mapTask function.
 * Used for filter-based query test mocks.
 */
export type MappedTask = {
    id: string
    content: string
    description: string
    dueDate: string | undefined
    recurring: string | boolean
    priority: number
    projectId: string
    sectionId: string | null
    parentId: string | null
    labels: string[]
}

/**
 * Creates a mock Task with all required properties and sensible defaults.
 * Pass only the properties you want to override for your specific test.
 */
export function createMockTask(overrides: Partial<Task> = {}): Task {
    return {
        id: '8485093748',
        content: 'Test task content',
        description: '',
        completedAt: null,
        labels: [],
        childOrder: 1,
        priority: 1,
        projectId: '6cfCcrrCFg2xP94Q',
        sectionId: null,
        parentId: null,
        url: 'https://todoist.com/showTask?id=8485093748',
        // Use correct property names from Task schema
        noteCount: 0,
        addedByUid: '713437',
        addedAt: '2025-08-13T22:09:56.123456Z',
        deadline: null,
        responsibleUid: null,
        assignedByUid: null,
        isCollapsed: false,
        isDeleted: false,
        duration: null,
        checked: false,
        updatedAt: '2025-08-13T22:09:56.123456Z',
        due: null,
        dayOrder: 0,
        userId: '713437',
        ...overrides,
    }
}

/**
 * Creates a mock Section with all required properties and sensible defaults.
 * Pass only the properties you want to override for your specific test.
 */
export function createMockSection(overrides: Partial<Section> = {}): Section {
    return {
        id: 'section-123',
        projectId: '6cfCcrrCFg2xP94Q',
        sectionOrder: 1,
        userId: 'test-user',
        addedAt: '2024-01-01T00:00:00Z',
        updatedAt: '2024-01-01T00:00:00Z',
        archivedAt: null,
        isArchived: false,
        isDeleted: false,
        isCollapsed: false,
        name: 'Test Section',
        ...overrides,
    }
}

/**
 * Creates a mock PersonalProject with all required properties and sensible defaults.
 * Pass only the properties you want to override for your specific test.
 */
export function createMockProject(overrides: Partial<PersonalProject> = {}): PersonalProject {
    return {
        id: '6cfCcrrCFg2xP94Q',
        name: 'Test Project',
        color: 'charcoal',
        isFavorite: false,
        isShared: false,
        parentId: null,
        inboxProject: false,
        viewStyle: 'list',
        url: 'https://todoist.com/projects/6cfCcrrCFg2xP94Q',
        isDeleted: false,
        updatedAt: '2025-08-13T22:09:55.841800Z',
        createdAt: '2025-08-13T22:09:55.841785Z',
        childOrder: 1,
        defaultOrder: 0,
        description: '',
        isCollapsed: false,
        canAssignTasks: false,
        isFrozen: false,
        isArchived: false,
        ...overrides,
    }
}

/**
 * Creates a mock API response object with results and nextCursor.
 */
export function createMockApiResponse<T>(
    results: T[],
    nextCursor: string | null = null,
): {
    results: T[]
    nextCursor: string | null
} {
    return {
        results,
        nextCursor,
    }
}

/**
 * Creates a simplified mapped task (matches mapTask output) for filter-based query tests.
 */
export function createMappedTask(overrides: Partial<MappedTask> = {}): MappedTask {
    return {
        id: TEST_IDS.TASK_1,
        content: 'Test task content',
        description: '',
        dueDate: undefined,
        recurring: false,
        priority: 1,
        projectId: TEST_IDS.PROJECT_TEST,
        sectionId: null,
        parentId: null,
        labels: [],
        ...overrides,
    }
}

/**
 * Common error messages used across tests.
 */
export const TEST_ERRORS = {
    API_RATE_LIMIT: 'API Error: Rate limit exceeded',
    API_UNAUTHORIZED: 'API Error: Unauthorized',
    INVALID_CURSOR: 'Invalid cursor format',
    INVALID_FILTER: 'Invalid filter query',
} as const

/**
 * Creates multiple test cases for parameterized testing.
 */
export function createTestCases<T, E = unknown>(
    cases: Array<{ name: string; input: T; expected?: E }>,
) {
    return cases
}

/**
 * Common mock IDs used across tests for consistency.
 */
export const TEST_IDS = {
    TASK_1: '8485093748',
    TASK_2: '8485093749',
    TASK_3: '8485093750',
    PROJECT_INBOX: 'inbox-project-id',
    PROJECT_WORK: 'work-project-id',
    PROJECT_TEST: '6cfCcrrCFg2xP94Q',
    SECTION_1: 'section-123',
    SECTION_2: 'section-456',
    USER_ID: '713437',
} as const
