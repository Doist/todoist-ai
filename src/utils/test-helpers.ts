import type { PersonalProject, Section, Task } from '@doist/todoist-api-typescript'
import { getToolOutput } from '../mcp-helpers'

/**
 * Mapped task type matching the output of mapTask function.
 * Used for filter-based query test mocks.
 */
export type MappedTask = {
    id: string
    content: string
    description: string
    dueDate: string | undefined
    recurring: string | boolean
    priority: number
    projectId: string
    sectionId: string | null
    parentId: string | null
    labels: string[]
    duration: string | null
    responsibleUid: string | null
    assignedByUid: string | null
}

/**
 * Creates a mock Task with all required properties and sensible defaults.
 * Pass only the properties you want to override for your specific test.
 */
export function createMockTask(overrides: Partial<Task> = {}): Task {
    return {
        id: '8485093748',
        content: 'Test task content',
        description: '',
        completedAt: null,
        labels: [],
        childOrder: 1,
        priority: 1,
        projectId: '6cfCcrrCFg2xP94Q',
        sectionId: null,
        parentId: null,
        url: 'https://todoist.com/showTask?id=8485093748',
        // Use correct property names from Task schema
        noteCount: 0,
        addedByUid: '713437',
        addedAt: '2025-08-13T22:09:56.123456Z',
        deadline: null,
        responsibleUid: null,
        assignedByUid: null,
        isCollapsed: false,
        isDeleted: false,
        duration: null,
        checked: false,
        updatedAt: '2025-08-13T22:09:56.123456Z',
        due: null,
        dayOrder: 0,
        userId: '713437',
        ...overrides,
    }
}

/**
 * Creates a mock Section with all required properties and sensible defaults.
 * Pass only the properties you want to override for your specific test.
 */
export function createMockSection(overrides: Partial<Section> = {}): Section {
    return {
        id: 'section-123',
        projectId: '6cfCcrrCFg2xP94Q',
        sectionOrder: 1,
        userId: 'test-user',
        addedAt: '2024-01-01T00:00:00Z',
        updatedAt: '2024-01-01T00:00:00Z',
        archivedAt: null,
        isArchived: false,
        isDeleted: false,
        isCollapsed: false,
        name: 'Test Section',
        url: 'https://todoist.com/sections/section-123',
        ...overrides,
    }
}

/**
 * Creates a mock PersonalProject with all required properties and sensible defaults.
 * Pass only the properties you want to override for your specific test.
 */
export function createMockProject(overrides: Partial<PersonalProject> = {}): PersonalProject {
    return {
        id: '6cfCcrrCFg2xP94Q',
        name: 'Test Project',
        color: 'charcoal',
        isFavorite: false,
        isShared: false,
        parentId: null,
        inboxProject: false,
        viewStyle: 'list',
        url: 'https://todoist.com/projects/6cfCcrrCFg2xP94Q',
        isDeleted: false,
        updatedAt: '2025-08-13T22:09:55.841800Z',
        createdAt: '2025-08-13T22:09:55.841785Z',
        childOrder: 1,
        defaultOrder: 0,
        description: '',
        isCollapsed: false,
        canAssignTasks: false,
        isFrozen: false,
        isArchived: false,
        ...overrides,
    }
}

/**
 * Creates a mock API response object with results and nextCursor.
 */
export function createMockApiResponse<T>(
    results: T[],
    nextCursor: string | null = null,
): {
    results: T[]
    nextCursor: string | null
} {
    return {
        results,
        nextCursor,
    }
}

/**
 * Creates a simplified mapped task (matches mapTask output) for filter-based query tests.
 */
export function createMappedTask(overrides: Partial<MappedTask> = {}): MappedTask {
    return {
        id: TEST_IDS.TASK_1,
        content: 'Test task content',
        description: '',
        dueDate: undefined,
        recurring: false,
        priority: 1,
        projectId: TEST_IDS.PROJECT_TEST,
        sectionId: null,
        parentId: null,
        labels: [],
        duration: null,
        responsibleUid: null,
        assignedByUid: null,
        ...overrides,
    }
}

/**
 * Common error messages used across tests.
 */
export const TEST_ERRORS = {
    API_RATE_LIMIT: 'API Error: Rate limit exceeded',
    API_UNAUTHORIZED: 'API Error: Unauthorized',
    INVALID_CURSOR: 'Invalid cursor format',
    INVALID_FILTER: 'Invalid filter query',
} as const

/**
 * Creates multiple test cases for parameterized testing.
 */
export function createTestCases<T, E = unknown>(
    cases: Array<{ name: string; input: T; expected?: E }>,
) {
    return cases
}

/**
 * Extracts the text content from a tool output for snapshot testing.
 * This allows tests to match against just the text content while tools return structured output.
 */
export function extractTextContent(toolOutput: unknown): string {
    if (typeof toolOutput === 'string') {
        return toolOutput
    }

    if (typeof toolOutput === 'object' && toolOutput !== null && 'content' in toolOutput) {
        const output = toolOutput as { content: unknown }
        if (
            Array.isArray(output.content) &&
            output.content[0] &&
            typeof output.content[0] === 'object' &&
            output.content[0] !== null &&
            'type' in output.content[0] &&
            'text' in output.content[0] &&
            output.content[0].type === 'text'
        ) {
            return output.content[0].text as string
        }
    }

    throw new Error('Expected tool output to have text content')
}

/**
 * Extracts the structured content from a tool output for testing.
 * This handles both the new `structuredContent` field and legacy JSON-encoded content.
 */
export function extractStructuredContent(
    output: ReturnType<typeof getToolOutput>,
): Record<string, unknown> {
    // Check for new structuredContent field first
    if ('structuredContent' in output && typeof output.structuredContent === 'object') {
        return output.structuredContent as Record<string, unknown>
    }

    // Fall back to checking for JSON content in the content array
    if ('content' in output && Array.isArray(output.content)) {
        for (const item of output.content) {
            if (
                typeof item === 'object' &&
                item !== null &&
                'type' in item &&
                'text' in item &&
                item.type === 'text' &&
                item.mimeType === 'application/json'
            ) {
                return JSON.parse(item.text) as Record<string, unknown>
            }
        }
    }

    throw new Error('Expected tool output to have structured content')
}

/**
 * Common mock IDs used across tests for consistency.
 */
export const TEST_IDS = {
    TASK_1: '8485093748',
    TASK_2: '8485093749',
    TASK_3: '8485093750',
    PROJECT_INBOX: 'inbox-project-id',
    PROJECT_WORK: 'work-project-id',
    PROJECT_TEST: '6cfCcrrCFg2xP94Q',
    SECTION_1: 'section-123',
    SECTION_2: 'section-456',
    USER_ID: '713437',
} as const

/**
 * Fixed date for consistent test snapshots.
 * Use this instead of new Date() in tests to avoid snapshot drift.
 */
export const TODAY = '2025-08-17' as const
