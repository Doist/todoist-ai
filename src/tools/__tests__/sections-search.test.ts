import type { Section, TodoistApi } from '@doist/todoist-api-typescript'
import { jest } from '@jest/globals'
import { sectionsSearch } from '../sections-search'
import { TEST_ERRORS, TEST_IDS, createMockSection } from '../test-helpers'

// Mock the Todoist API
const mockTodoistApi = {
    getSections: jest.fn(),
} as unknown as jest.Mocked<TodoistApi>

describe('sections-search tool', () => {
    beforeEach(() => {
        jest.clearAllMocks()
    })

    describe('listing all sections in a project', () => {
        it('should list all sections when no search parameter is provided', async () => {
            const mockSections: Section[] = [
                createMockSection({
                    id: TEST_IDS.SECTION_1,
                    projectId: TEST_IDS.PROJECT_TEST,
                    name: 'To Do',
                }),
                createMockSection({
                    id: TEST_IDS.SECTION_2,
                    projectId: TEST_IDS.PROJECT_TEST,
                    sectionOrder: 2,
                    name: 'In Progress',
                }),
                createMockSection({
                    id: 'section-789',
                    projectId: TEST_IDS.PROJECT_TEST,
                    sectionOrder: 3,
                    name: 'Done',
                }),
                createMockSection({
                    id: 'section-999',
                    projectId: TEST_IDS.PROJECT_TEST,
                    sectionOrder: 4,
                    name: 'Backlog Items',
                }),
            ]

            mockTodoistApi.getSections.mockResolvedValue({
                results: mockSections,
                nextCursor: null,
            })

            const result = await sectionsSearch.execute(
                { projectId: TEST_IDS.PROJECT_TEST },
                mockTodoistApi,
            )

            // Verify API was called correctly
            expect(mockTodoistApi.getSections).toHaveBeenCalledWith({
                projectId: TEST_IDS.PROJECT_TEST,
            })

            // Verify result is properly mapped (simplified format)
            expect(result).toEqual([
                { id: TEST_IDS.SECTION_1, name: 'To Do' },
                { id: TEST_IDS.SECTION_2, name: 'In Progress' },
                { id: 'section-789', name: 'Done' },
                { id: 'section-999', name: 'Backlog Items' },
            ])
        })

        it('should handle project with no sections', async () => {
            mockTodoistApi.getSections.mockResolvedValue({
                results: [],
                nextCursor: null,
            })

            const result = await sectionsSearch.execute(
                { projectId: 'empty-project-id' },
                mockTodoistApi,
            )

            expect(mockTodoistApi.getSections).toHaveBeenCalledWith({
                projectId: 'empty-project-id',
            })

            expect(result).toEqual([])
        })
    })

    describe('searching sections by name', () => {
        it('should filter sections by search term (case insensitive)', async () => {
            const mockSections: Section[] = [
                createMockSection({
                    id: TEST_IDS.SECTION_1,
                    projectId: TEST_IDS.PROJECT_TEST,
                    name: 'To Do',
                }),
                createMockSection({
                    id: TEST_IDS.SECTION_2,
                    projectId: TEST_IDS.PROJECT_TEST,
                    sectionOrder: 2,
                    name: 'In Progress',
                }),
                createMockSection({
                    id: 'section-789',
                    projectId: TEST_IDS.PROJECT_TEST,
                    sectionOrder: 3,
                    name: 'Done',
                }),
                createMockSection({
                    id: 'section-999',
                    projectId: TEST_IDS.PROJECT_TEST,
                    sectionOrder: 4,
                    name: 'Progress Review',
                }),
            ]

            mockTodoistApi.getSections.mockResolvedValue({
                results: mockSections,
                nextCursor: null,
            })

            const result = await sectionsSearch.execute(
                {
                    projectId: TEST_IDS.PROJECT_TEST,
                    search: 'progress',
                },
                mockTodoistApi,
            )

            expect(mockTodoistApi.getSections).toHaveBeenCalledWith({
                projectId: TEST_IDS.PROJECT_TEST,
            })

            // Should return both "In Progress" and "Progress Review" (case insensitive partial match)
            expect(result).toEqual([
                { id: TEST_IDS.SECTION_2, name: 'In Progress' },
                { id: 'section-999', name: 'Progress Review' },
            ])
        })

        it('should handle search with no matches', async () => {
            const mockSections: Section[] = [
                createMockSection({
                    id: TEST_IDS.SECTION_1,
                    projectId: TEST_IDS.PROJECT_TEST,
                    name: 'To Do',
                }),
                createMockSection({
                    id: TEST_IDS.SECTION_2,
                    projectId: TEST_IDS.PROJECT_TEST,
                    sectionOrder: 2,
                    name: 'In Progress',
                }),
            ]

            mockTodoistApi.getSections.mockResolvedValue({
                results: mockSections,
                nextCursor: null,
            })

            const result = await sectionsSearch.execute(
                {
                    projectId: TEST_IDS.PROJECT_TEST,
                    search: 'nonexistent',
                },
                mockTodoistApi,
            )

            expect(result).toEqual([])
        })

        it('should handle case sensitive search correctly', async () => {
            const mockSections: Section[] = [
                createMockSection({
                    id: TEST_IDS.SECTION_1,
                    projectId: TEST_IDS.PROJECT_TEST,
                    name: 'Important Tasks',
                }),
                createMockSection({
                    id: TEST_IDS.SECTION_2,
                    projectId: TEST_IDS.PROJECT_TEST,
                    sectionOrder: 2,
                    name: 'Regular Work',
                }),
            ]

            mockTodoistApi.getSections.mockResolvedValue({
                results: mockSections,
                nextCursor: null,
            })

            const result = await sectionsSearch.execute(
                {
                    projectId: TEST_IDS.PROJECT_TEST,
                    search: 'IMPORTANT',
                },
                mockTodoistApi,
            )

            // Should match despite different case
            expect(result).toHaveLength(1)
            expect(result[0]).toEqual({ id: TEST_IDS.SECTION_1, name: 'Important Tasks' })
        })

        it('should handle partial matches correctly', async () => {
            const mockSections: Section[] = [
                createMockSection({
                    id: TEST_IDS.SECTION_1,
                    projectId: TEST_IDS.PROJECT_TEST,
                    name: 'Development Tasks',
                }),
                createMockSection({
                    id: TEST_IDS.SECTION_2,
                    projectId: TEST_IDS.PROJECT_TEST,
                    sectionOrder: 2,
                    name: 'Testing Tasks',
                }),
                createMockSection({
                    id: 'section-789',
                    projectId: TEST_IDS.PROJECT_TEST,
                    sectionOrder: 3,
                    name: 'Deployment',
                }),
            ]

            mockTodoistApi.getSections.mockResolvedValue({
                results: mockSections,
                nextCursor: null,
            })

            const result = await sectionsSearch.execute(
                {
                    projectId: TEST_IDS.PROJECT_TEST,
                    search: 'task',
                },
                mockTodoistApi,
            )

            // Should match both sections with "task" in the name
            expect(result).toEqual([
                { id: TEST_IDS.SECTION_1, name: 'Development Tasks' },
                { id: TEST_IDS.SECTION_2, name: 'Testing Tasks' },
            ])
        })

        it('should handle exact matches', async () => {
            const mockSections: Section[] = [
                createMockSection({
                    id: TEST_IDS.SECTION_1,
                    projectId: TEST_IDS.PROJECT_TEST,
                    name: 'Done',
                }),
                createMockSection({
                    id: TEST_IDS.SECTION_2,
                    projectId: TEST_IDS.PROJECT_TEST,
                    sectionOrder: 2,
                    name: 'Done Soon',
                }),
            ]

            mockTodoistApi.getSections.mockResolvedValue({
                results: mockSections,
                nextCursor: null,
            })

            const result = await sectionsSearch.execute(
                {
                    projectId: TEST_IDS.PROJECT_TEST,
                    search: 'done',
                },
                mockTodoistApi,
            )

            // Should match both sections containing "done"
            expect(result).toEqual([
                { id: TEST_IDS.SECTION_1, name: 'Done' },
                { id: TEST_IDS.SECTION_2, name: 'Done Soon' },
            ])
        })
    })

    describe('error handling', () => {
        it.each([
            { error: 'API Error: Project not found', projectId: 'non-existent-project' },
            { error: TEST_ERRORS.API_UNAUTHORIZED, projectId: 'restricted-project' },
            { error: 'API Error: Invalid project ID format', projectId: 'invalid-id-format' },
        ])('should propagate $error', async ({ error, projectId }) => {
            mockTodoistApi.getSections.mockRejectedValue(new Error(error))
            await expect(sectionsSearch.execute({ projectId }, mockTodoistApi)).rejects.toThrow(
                error,
            )
        })
    })
})
