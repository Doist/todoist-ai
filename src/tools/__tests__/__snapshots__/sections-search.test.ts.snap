// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`sections-search tool listing all sections in a project should handle project with no sections 1`] = `
"Sections in project empty-project-id: 0.
No results. Project has no sections yet; Use sections-manage to create sections."
`;

exports[`sections-search tool listing all sections in a project should list all sections when no search parameter is provided 1`] = `
"Sections in project 6cfCcrrCFg2xP94Q: 4.
Preview:
    To Do • id=section-123
    In Progress • id=section-456
    Done • id=section-789
    Backlog Items • id=section-999
Next:
- Use tasks-list-for-container with type=section to see tasks in any section
- Use sections-manage to modify section names or order"
`;

exports[`sections-search tool searching sections by name should filter sections by search term (case insensitive) 1`] = `
"Sections in project 6cfCcrrCFg2xP94Q matching "progress": 2.
Preview:
    In Progress • id=section-456
    Progress Review • id=section-999
Next:
- Use tasks-list-for-container with type=section to see tasks in any section
- Use sections-manage to modify section names or order
- Remove search parameter to see all sections in this project"
`;

exports[`sections-search tool searching sections by name should handle case sensitive search correctly 1`] = `
"Sections in project 6cfCcrrCFg2xP94Q matching "IMPORTANT": 1.
Preview:
    Important Tasks • id=section-123
Next:
- Use tasks-list-for-container with type=section and id=section-123 to see tasks
- Use sections-manage to create additional sections for organization
- Remove search parameter to see all sections in this project"
`;

exports[`sections-search tool searching sections by name should handle exact matches 1`] = `
"Sections in project 6cfCcrrCFg2xP94Q matching "done": 2.
Preview:
    Done • id=section-123
    Done Soon • id=section-456
Next:
- Use tasks-list-for-container with type=section to see tasks in any section
- Use sections-manage to modify section names or order
- Remove search parameter to see all sections in this project"
`;

exports[`sections-search tool searching sections by name should handle partial matches correctly 1`] = `
"Sections in project 6cfCcrrCFg2xP94Q matching "task": 2.
Preview:
    Development Tasks • id=section-123
    Testing Tasks • id=section-456
Next:
- Use tasks-list-for-container with type=section to see tasks in any section
- Use sections-manage to modify section names or order
- Remove search parameter to see all sections in this project"
`;

exports[`sections-search tool searching sections by name should handle search with no matches 1`] = `
"Sections in project 6cfCcrrCFg2xP94Q matching "nonexistent": 0.
No results. Try broader search terms; Check spelling; Remove search to see all sections."
`;
